tests/unit/services/test_instrument_conversion_service.py

```diff
@@ -0,0 +1,189 @@
+    "environment_variables",
+    [
+        # No URL
+        ({}),
+        # Empty URL
+        ({"CONVERTER_SERVICE_API_BASE_URL": ""}),
+    ],
+)
+async def test_retrieve_instrument_missing_converter_base_url(environment_variables, mocker):
+    """Test convert_instrument raises HTTPException if CONVERTER_SERVICE_API_BASE_URL is missing or empty."""
+    instrument = {"id": "123", "validator_version": "1.0.0", "sections": []}
+    target_version = "2.0.0"
+    # Patch environment to remove CONVERTER_SERVICE_API_BASE_URL
+    mocker.patch.dict(os.environ, environment_variables, clear=True)
+    with pytest.raises(HTTPException) as exc_info:
+        await convert_instrument(instrument, target_version)
+    exc = exc_info.value
+    assert exc.status_code == 500
+    assert exc.detail["message"] == "CONVERTER_SERVICE_API_BASE_URL configuration is missing."
+
+
+@pytest.mark.asyncio
+async def test_retrieve_instrument_missing_converter_endpoint(mocker):
+    """Test convert_instrument raises HTTPException if CONVERTER_SERVICE_CONVERT_CI_ENDPOINT exists but has no value."""
+    instrument = {"id": "123", "validator_version": "1.0.0", "sections": []}
+    target_version = "2.0.0"
+    # Patch environment to include CONVERTER_SERVICE_API_BASE_URL but not CONVERTER_SERVICE_CONVERT_CI_ENDPOINT
+    mocker.patch.dict(
+        os.environ,
+        {"CONVERTER_SERVICE_API_BASE_URL": "http://fake-url", "CONVERTER_SERVICE_CONVERT_CI_ENDPOINT": ""},
+        clear=True,
+    )
+    with pytest.raises(HTTPException) as exc_info:
+        await convert_instrument(instrument, target_version)
+    exc = exc_info.value
+    assert exc.status_code == 500
+    assert exc.detail["message"] == "CONVERTER_SERVICE_CONVERT_CI_ENDPOINT configuration is missing."
+
+
+def test_safe_parse_valid(monkeypatch):
+    """Tests the safe_parse function with a valid version."""
+
+    class DummyVersion:  # pylint: disable=too-few-public-methods
+        """Dummy version parser for testing."""
+
+        @staticmethod
+        def parse(s: str):
+            """Parses a version string."""
+            return f"parsed-{s}"
+
+    monkeypatch.setattr("eq_cir_proxy_service.services.instrument.instrument_conversion_service.Version", DummyVersion)
+
+    result = safe_parse("current", "1.2.3")
+    assert result == "parsed-1.2.3"
+
+
+@pytest.mark.parametrize(
+    "version_type, version_value",
+    [
+        ("current", "abc"),
+        ("target", "???"),
+    ],
+)
+def test_safe_parse_invalid_version(version_type, version_value, monkeypatch):
+    """Tests the safe_parse function with an invalid version."""
+
+    class DummyVersion:  # pylint: disable=too-few-public-methods
+        """Dummy version parser for testing."""
+
+        @staticmethod
+        def parse(s: str):
+            """Parses a version string."""
+            error_message = "bad version"
+            if s == "good":
+                return "parsed-good"
+            raise ValueError(error_message)
```

Comment:
```suggestion
        def parse(s: str):
            """Parses a version string."""
            error_message = "invalid version"
            if s == "valid":
                return "parsed-valid"
            raise ValueError(error_message)
```
Renames to `valid` and `invalid`. Code comments will also require updates from this.